### ISSUE TITLE
problem on armv7 qnap ts-231p3

### ISSUE THREAD

**Issue by SiNaPsEr0x** [qbittorrent/docker-qbittorrent-nox#40]
after setting env etc.. error on run

<img width="654" alt="image" src="https://github.com/qbittorrent/docker-qbittorrent-nox/assets/5438823/e68cc28d-ba39-415d-98d0-76cbae5b052d">

**Comment by SiNaPsEr0x** [comment]
probably is boost library... on qnap is 1.84 but new qbit use a 1.85... is possible to force update this?

<img width="324" alt="image" src="https://github.com/qbittorrent/docker-qbittorrent-nox/assets/5438823/e80c7417-155a-456b-819b-5d7aac108931">

**Comment by Chocobo1** [comment]
Your issue look a lot like this: https://github.com/alpinelinux/docker-alpine/issues/342
And unfortunately it is not possible to resolve it in our project. Please report it to Alpine Linux: https://gitlab.alpinelinux.org/alpine/aports/-/issues

**Comment by SiNaPsEr0x** [comment]
Ts231 have a alpine 3.14.x but its same

**Comment by Chocobo1** [comment]
Or maybe the kernel mismatch is too great and causing [issues](https://stackoverflow.com/a/56606244).
Anyway... you can consider using static built qbt: https://github.com/userdocs/qbittorrent-nox-static

**Comment by SiNaPsEr0x** [comment]
<img width="682" alt="image" src="https://github.com/qbittorrent/docker-qbittorrent-nox/assets/5438823/f4c50dc8-0731-44c3-a1ab-b5180c7fc76b">

**Comment by Chocobo1** [comment]
>can you add qts for qnap?

You'll need to ask them for help, not here.

At this point I don't think there is more I can provide, closing issue.

**Comment by SiNaPsEr0x** [comment]
I prefer to use the armhf version because in addition to working it would seem to be better

[](https://unix.stackexchange.com/questions/751294/what-debian-arch-should-i-use-for-armv7l-kernel)

**Comment by SiNaPsEr0x** [comment]
ARMhf (ARM hard float):

Architecture: Based on ARMv7-A.
FPU Support: Supports floating point unit (FPU) with the VFPv3-D16 extension.
Objective: Optimized for the latest and most powerful devices.
Usage: Common in smartphones, tablets and embedded systems.
Software packages: Labeled as armhf.


ARMv7l:

Architecture: Also based on ARMv7-A.
FPU support: Does not support FPU (floating-point unit).
Target: Aimed at older or low-power devices.
Usage: Less common than ARMhf.
Software packages: Labeled as arm/armel.


In summary, if your ARM processor has an FPU, you should prefer ARMhf to get the best performance. If, however, you are working with older devices or without FPUs, you can opt for ARMv7l[1] [2].

**Comment by SiNaPsEr0x** [comment]
Wouldn't it be better to use since it works the HF version?

**Comment by SiNaPsEr0x** [comment]
I beg your pardon for asking but I just wanted to understand the differences -- what version is best to start on
